users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_configurable.py

```diff
@@ -177,29 +243,71 @@ def get_masked():
         )
         actual_freq_mask_max_num = tf.minimum(freq_mask_max_num, total_freq_masks_max_size // freq_mask_max_size)
 
-        x_masked = random_mask(
-            x_masked,
-            batch_axis=data.batch_dim_axis,
-            axis=data.time_dim_axis,
-            min_num=0,
-            max_num=actual_time_mask_max_num,
-            max_dims=time_mask_max_size,
-            sorted_indices=tf.range(tf.shape(x)[data.time_dim_axis]),
-        )
-        x_masked = random_mask(
-            x_masked,
-            batch_axis=data.batch_dim_axis,
-            axis=data.feature_dim_axis,
-            min_num=0,
-            max_num=actual_freq_mask_max_num,
-            max_dims=freq_mask_max_size,
-            sorted_indices=sorted_indices,
-        )
+        if config["filter_based_masking_strategy"] == "variance":
+            f_resp = get_frequency_response(filter_layer)
+            variance = tf.math.reduce_variance(f_resp, axis=0)
+            n_features = tf.shape(x)[data.feature_dim_axis]
+            probs = tf.nn.softmax(variance)
+            uniform_probs = tf.ones_like(probs) / tf.cast(n_features, tf.float32)
+            final_probs = variance_factor * probs + (1 - variance_factor) * uniform_probs
+            enable_logging = tf.convert_to_tensor(config["enable_logging"], dtype=tf.bool)
+
+            def logging_ops():
+                with tf.control_dependencies(
+                    [
+                        tf.print(
+                            "Specaug variance Log: ",
+                            current_epoch,
+                            actual_time_mask_max_num,
+                            tf.shape(x)[data.time_dim_axis],
+                            variance.shape,
+                            sep=", ",
+                        )
+                    ]
+                ):
+                    return tf.identity(x_masked)
+
+            x_masked = tf.cond(enable_logging, logging_ops, lambda: tf.identity(x_masked))
```

Comment:
Shouldn't this be in `filter_based_masking` where the actual mask size is determined?

Reply:
What I meant was you can do a separate PR to change what currently exists. I didn't think of adding the same old behavior here. So I would add it with the actual masks directly here but I don't mind if you keep it like this for now.